File: selection_sort/Selectionsort.go
package main

import "fmt"

func main() {
	sorted_arr := selection_sort([]int{31, 34, 45, 23, 55, 12, 14, 98, 21})
	fmt.Println(sorted_arr)
}

func findSmallest(arr []int) int {
	// smallest_index := 0
	smallest_number := arr[0]
	for _, v := range arr {
		if v < smallest_number {
			smallest_number = v
			// smallest_index = i

		}
	}
	return smallest_number
}

func selection_sort(arr []int) []int {
	new_arr := []int{}
	for range arr {
		smallest := findSmallest(arr)
		new_arr = append(new_arr, smallest)
		arr = remove(arr, smallest)
	}
	return new_arr
}
func remove(arr []int, n int) []int {
	for i, v := range arr {
		if v == n {
			arr = append(arr[:i], arr[i+1:]...)
		}
	}
	return arr
}

File: selection_sort/Selectionsort.py
def find_smallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index


def selection_sort(arr):
    new_arr = []
    for _ in range(len(arr)):
        smallest = find_smallest(arr)
        new_arr.append(arr.pop(smallest))
    return new_arr    

def main():
    print(selection_sort([31, 34 ,45, 23, 55, 12, 14, 98, 21]))
    


if __name__ == "__main__":
    main()
